### PR TITLE
New HandleJobsWithBucketNamePermission permission

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

**Comment by fviale** [comment]
Could you add some unit tests like HandleJobsWithGenericInformationPermissionTest ?

**Comment by fviale** [comment]
Additionnally new entries should be added in the "user" group in file config/security.java.policy-server
such as 
permission org.ow2.proactive.scheduler.permissions.HandleJobsWithGenericInformationPermission "";
by default it is empty, which means a member of "user" does not have a specific allowance.
This helps to have it at hand when we want to modify it for a customer

**Comment by Mykhailenko** [comment]
@fviale aight

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.ow2.proactive.permissions.ClientPermission;
+
+
+public class HandleJobsWithBucketNamePermission extends ClientPermission {
+    private final static String DESCRIPTION = "Handle Jobs that contains specific 'bucket_name' in Generic Information";
+
+    public final static String BUCKET_NAME = "bucket_name";
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java]: it is "bucketName"

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermission.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                                       .map(String::trim)
+                                       .collect(Collectors.toSet()));
+        }
+    }
+
+    // persmission of a job that needs to be authorised
+    public HandleJobsWithGroupNamePermission(Map<String, String> genericInformations) {
+        super(DESCRIPTION);
+        if (genericInformations.containsKey(GROUP)) {
+            String bucketName = genericInformations.get(GROUP);
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermission.java]: groupName

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        Map<String, String> genericInfo = new HashMap<>();
+        genericInfo.put(HandleJobsWithGroupNamePermission.GROUP, "quatro");
+
+        HandleJobsWithGroupNamePermission persmissionFromJob = new HandleJobsWithGroupNamePermission(genericInfo);
+
+        assertFalse(fromSecutiryFile.implies(persmissionFromJob));
+    }
+
+    @Test
+    public void testThatJobForbiden2Buckets() {
```
Comment (fviale) [inline scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java]: 2Groups

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+public class HandleJobsWithGroupNamePermissionTest {
+
+    @Test
+    public void testThatJobAllowed() {
+        HandleJobsWithGroupNamePermission fromSecutiryFile = new HandleJobsWithGroupNamePermission("uno,dos,trez");
+
+        Map<String, String> genericInfo = new HashMap<>();
+        genericInfo.put(HandleJobsWithGroupNamePermission.GROUP, "uno");
+
+        HandleJobsWithGroupNamePermission persmissionFromJob = new HandleJobsWithGroupNamePermission(genericInfo);
```
Comment (fviale) [inline scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java]: you could test all 3 items

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermissionTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+public class HandleJobsWithBucketNamePermissionTest {
+
+    @Test
+    public void testThatJobAllowed() {
+        HandleJobsWithBucketNamePermission fromSecutiryFile = new HandleJobsWithBucketNamePermission("uno,dos,trez");
+
+        Map<String, String> genericInfo = new HashMap<>();
+        genericInfo.put(HandleJobsWithBucketNamePermission.BUCKET_NAME, "uno");
+
+        HandleJobsWithBucketNamePermission persmissionFromJob = new HandleJobsWithBucketNamePermission(genericInfo);
```
Comment (fviale) [inline scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermissionTest.java]: you could test all 3 items

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.ow2.proactive.permissions.ClientPermission;
+
+
+public class HandleJobsWithBucketNamePermission extends ClientPermission {
+    private final static String DESCRIPTION = "Handle Jobs that contains specific 'bucket_name' in Generic Information";
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java]: contain

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java (hunk 1 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java]: ok

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermission.java (hunk 2 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermission.java]: sure

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java (hunk 3 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java]: aight

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java (hunk 4 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithGroupNamePermissionTest.java]: okok

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java (hunk 6 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermission.java]: thanks

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermissionTest.java (hunk 5 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/permissions/HandleJobsWithBucketNamePermissionTest.java]: sure

**Pull request merged**